### DISCUSSION TITLE
Calculate pore size distribution on part of the image domain

### DISCUSSION THREAD

**Discussion by thsmit** [PMEAL/porespy#833] [Q&A]
Hi all,

I have a 3D image and I would like to run analysis (pore size distribution) only on part of the 3D image domain. Similar to a segmentation of a CT-scan. Can anyone advice on a way to do this?

Thanks, Thijs

**Comment by jgostick** [comment]
Do you mean you just want to analyze a subset, like:

im2 = im[:100, :100, :100] 

This is just a standard numpy indexing notation.

**Reply by thsmit** [reply]
Thanks for your quick reply. 

Yes, I mean a subset, but with a more 'complex' shape. Like you would have if you have a segmentation of a CT scan. For example by full domain is an image im.shape =(500,500,500) and the subset is defined with a vector 0, 1, 1, 0, ... (with a shape of a subset that make sense).

**Reply by jgostick** [reply]
I thought this question was posted on the porespy site, but see now that you're asking about openpnm.  There is not preset way to select coords, but you can use the `find_nearby_pores` method.  So given a pore at the center of the domain, you can find all pores within R units using ``pn.find_nearby_pores(pores=P, R=R)`` where P is the pore index of the center pore and R is your desired radius.  You could make a cylinder with ``Ps = pn.coords[10, 10, :] == 5``, then use the above method.  Is that what you're looking for?

**Reply by thsmit** [reply]
Yes, that could be a solution. 

I was thinking of pre-setting all the voxels outside of the subdomain 1/True, meaning there are no pores outside of the subset that should be analyzed. I will test it out.

**Reply by jgostick** [reply]
I'm still not quite sure what you're hoping to do with this, but the method I mention above is how to identify pores in certain locations.   You would then presumably use the trim method to delete them before running simulations on the subsections.

**Reply by thsmit** [reply]
For example, I would like to analysis the pore size and strut size distribution of this porous femur. But the femur is embedded in a grid, larger than the femur, so the grid has inactive voxels around the femur. 

![femur_porous](https://github.com/PMEAL/OpenPNM/assets/52911749/44471e55-1e84-47ef-8cf7-7bf39593b39c)

**Reply by jgostick** [reply]
This seems more like a PoreSpy problem.  You have the femur in an image, like tiff format, I presume.  So you open in PoreSpy, set solid to ones, and void to zeros, then you can analyze strut size with local thickness etc.  To analyze the cavities, you'd invert the image, but I guess this is where your problem lies, in differentiating the cavities inside the femur from the region outside, since they'll both be ones.  If the cavities are totally contained inside the femur, then you could find them by labeling the voids (scipy.ndimage.label) and note which of the labels corresponds to the region outside the femur.  For instance if this is labeled "1", then pores are `pores = labels > 1`.

**Reply by thsmit** [reply]
Thanks a lot, I can work with this!
